pkg/expansion/system.go

```diff
@@ -232,7 +228,20 @@ func expandResource(obj *unstructured.Unstructured, ns *corev1.Namespace, templa
 	resource := &unstructured.Unstructured{}
 	resource.SetUnstructuredContent(src)
 	resource.SetGroupVersionKind(resultantGVK)
-	resource.SetNamespace(ns.Name)
+	if ns != nil {
+		resource.SetNamespace(ns.Name)
+	} else {
+		nsFromUn, ok, err := unstructured.NestedString(obj.Object, "metadata", "namespace")
+		if err != nil {
+			return nil, fmt.Errorf("could not extract namespace field %q in resource %s", srcPath, obj.GetName())
+		}
+		if !ok {
```

Comment:
can we add a test case that covers this?

Reply:
n/a after latest commit